R/woods_plot.R

```diff
@@ -69,86 +71,185 @@ woods_plot <- function(data, fold_change, start_position, end_position, protein_
       dplyr::mutate({{ protein_id }} := paste0({{ protein_id }}, " (", round({{ coverage }}, digits = 1), "%)"))
   }
   # Create plot
-  data %>%
-    ggplot2::ggplot() +
-    ggplot2::geom_rect(ggplot2::aes(
-      xmin = 0,
-      xmax = {{ protein_length }},
-      ymin = -0.01,
-      ymax = 0.01
-    ),
-    fill = "black"
-    ) +
-    ggplot2::geom_rect(ggplot2::aes(
-      xmin = {{ start_position }},
-      xmax = {{ end_position }},
-      ymin = {{ fold_change }} - 0.5,
-      ymax = {{ fold_change }} + 0.5,
-      fill = {{ colouring }}
-    ),
-    col = "black",
-    size = 0.7
-    ) +
-    ggplot2::geom_hline(
-      yintercept = -{{ fold_change_cutoff }},
-      col = "blue",
-      alpha = .8,
-      size = 0.7
-    ) +
-    ggplot2::geom_hline(
-      yintercept = {{ fold_change_cutoff }},
-      col = "blue",
-      alpha = .8,
-      size = 0.7
-    ) +
-    ggplot2::ylim(min(-2.5, dplyr::pull(data, {{ fold_change }})) - 0.5, max(2.5, dplyr::pull(data, {{ fold_change }})) + 0.5) +
-    ggplot2::scale_x_continuous(limits = NULL, expand = c(0, 0)) +
-    ggplot2::labs(
-      x = "Protein Sequence",
-      y = "log2(fold change)",
-      title = {
-        if (!missing(protein_id)) unique(dplyr::pull(data, {{ protein_id }}))
-      }
-    ) +
-    {
-      if (!missing(facet)) ggplot2::facet_wrap(rlang::new_formula(NULL, rlang::enquo(facet)), scales = "free", ncol = 4)
-    } +
-    ggplot2::guides(size = FALSE) +
-    {
-      if (!missing(colouring) && !is.numeric(dplyr::pull(data, {{ colouring }}))) ggplot2::scale_fill_manual(values = protti_colours)
-    } +
-    {
-      if (!missing(colouring) && is.numeric(dplyr::pull(data, {{ colouring }}))) ggplot2::scale_fill_gradient(low = protti_colours[1], high = protti_colours[2])
-    } +
-    ggplot2::theme_bw() +
-    ggplot2::theme(
-      plot.title = ggplot2::element_text(
-        size = 20
+  if(!missing(highlight)) {
+    if(!all(is.logical(dplyr::pull(data, {{ highlight }} )))){
+      warning(paste("Please only provide logicals (i.e. TRUE or FALSE) in the 'highlight' column."))
```

Comment:
```suggestion
      stop("Please only provide logicals (i.e. TRUE or FALSE) in the 'highlight' column.")
```

Would make this an error. If the user supplies this wrong then no asterisks will be made so they should get an error that lets them properly know they did something wrong.